step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -272,7 +272,21 @@ public String getDisplayName() {
      * A type of a parameter to a class.
      */
     public static abstract class ParameterType {
-        ParameterType() {}
+        @Nonnull
+        private final Type actualType;
+
+        public Type getActualType() {
+            return actualType;
+        }
+
+        public Class<?> getType() {
+            return (Class) getActualType();
```

Comment:
Well, the `ArrayType` I would expect to have a `ParameterizedType` here, so casting to `Class` would throw `ClassCastException`.